internal/cmd/providers.go

```diff
@@ -245,6 +249,26 @@ $ infra providers add google --url accounts.google.com --client-id 0oa3sz06o6do0
 			}
 
 			cli.Output("Connected provider %q (%s) to infra", args[0], opts.URL)
+
+			if opts.SCIM {
+				key, err := client.CreateAccessKey(&api.CreateAccessKeyRequest{
+					UserID:            provider.ID,
```

Comment:
This call to create access key , and the one below in `updateProvider` both set the `ProviderID` to the infra provider, right?  

Could we set the `ProviderID` on this key to the scim provider as well? That would give us an easy way of finding these access keys that are for provider scim without having to first do the user lookup and fallback to the provider lookup.

Reply:
The ProviderID here indicates who granted the access key so we can' use it to prevent the fallback lookup. For example if a user signed in using Okta the provider ID would the Okta provider's ID, but we don't want them to be able to provision users within the Okta provider.

Reply:
We can't use that field on its own, but I was wondering if we could do something like:

```go
if accessKey.ProviderID == accessKey.IssusedFor { ... }
```

That gives us a way to identify the skim keys without having to compare the row to the user table. 
* any key where the providerId and issusedFor match is a scim key
* any key where they do not match is a regular user key

Otherwise we have to compare the field to another table to determine what kind of key it is.

Reply:
Clever, I'll add this

Reply:
Made this change:
https://github.com/infrahq/infra/pull/3407/commits/1e5d87d69eb362b6fbb4a3ef2b8126b609874ef2